src/test/java/it/gov/pagopa/reward/controller/RewardTrxSynchronousApiControllerIntegrationTest.java

```diff
@@ -142,11 +149,29 @@ private void publishRewardRules() {
                 .initiativeRewardType(InitiativeRewardType.DISCOUNT)
                 .build();
         rule.getGeneral().setBeneficiaryBudget(beneficiaryBudget);
-        kafkaTestUtilitiesService.publishIntoEmbeddedKafka(topicRewardRuleConsumer, null, null, rule);
-        RewardRuleConsumerConfigTest.waitForKieContainerBuild(1, rewardContextHolderService);
+
+        InitiativeReward2BuildDTO ruleNF = InitiativeReward2BuildDTOFaker.mockInstanceBuilder(0, Collections.emptySet(), RewardValueDTO.class)
+                .initiativeId(INITIATIVEID_NF)
+                .initiativeName("NAME_" + INITIATIVEID_NF)
+                .organizationId("ORGANIZATIONID_" + INITIATIVEID_NF)
+                .trxRule(InitiativeTrxConditions.builder()
+                        .threshold(ThresholdDTO.builder()
+                                .from(BigDecimal.valueOf(5))
+                                .fromIncluded(true)
+                                .build())
+                        .build())
+                .rewardRule(RewardValueDTO.builder()
+                        .rewardValue(BigDecimal.TEN)
+                        .build())
+                .initiativeRewardType(InitiativeRewardType.DISCOUNT)
+                .build();
+        ruleNF.getGeneral().setBeneficiaryBudget(beneficiaryBudget);
+        ruleNF.getGeneral().setBeneficiaryType(InitiativeGeneralDTO.BeneficiaryTypeEnum.NF);
+        List.of(rule, ruleNF).forEach(r -> kafkaTestUtilitiesService.publishIntoEmbeddedKafka(topicRewardRuleConsumer, null, null, r));
+        RewardRuleConsumerConfigTest.waitForKieContainerBuild(2, rewardContextHolderService);
```

Comment:
use empty line to make more readable the code, separating blocks by purpose
```suggestion
        ruleNF.getGeneral().setBeneficiaryType(InitiativeGeneralDTO.BeneficiaryTypeEnum.NF);
        
        List.of(rule, ruleNF).forEach(r -> kafkaTestUtilitiesService.publishIntoEmbeddedKafka(topicRewardRuleConsumer, null, null, r));
        RewardRuleConsumerConfigTest.waitForKieContainerBuild(2, rewardContextHolderService);
```